go/tendermint/abci/mux.go

```diff
@@ -363,13 +363,47 @@ func (mux *abciMux) InitChain(req types.RequestInitChain) types.ResponseInitChai
 	return resp
 }
 
+func (s *ApplicationState) InHaltEpoch() bool {
+	epoch, err := s.GetEpoch()
+	if err != nil {
+		s.logger.Error("InHaltEpoch: failed to get current epoch",
+			"err", err,
+		)
+		return false
+	}
+
+	return epoch == epochtime.EpochTime(s.haltEpochHeight)
+}
+
+func (s *ApplicationState) AfterHaltEpoch() bool {
+	epoch, err := s.GetEpoch()
+	if err != nil {
+		s.logger.Error("AfterHaltEpoch: failed to get current epoch",
+			"err", err,
+		)
+		return false
+	}
+	return epoch > epochtime.EpochTime(s.haltEpochHeight)
+}
+
 func (mux *abciMux) BeginBlock(req types.RequestBeginBlock) types.ResponseBeginBlock {
 	blockHeight := mux.state.BlockHeight()
 	mux.logger.Debug("BeginBlock",
 		"req", req,
 		"block_height", blockHeight,
 	)
 
+	if mux.state.InHaltEpoch() {
+		mux.logger.Info("BeginBlock: in halt epoch, emitting empty block.")
```

Comment:
There can be lots of blocks in the halt epoch, this will be ridiculously spammy and likely not useful.  It should log once and only once, on the transition.

I would like to see the block number and epoch logged as well, if we are doing this.

Reply:
makes sense :+1: will switch to only logging it once